### ISSUE TITLE
CLI: metadata cache ignores adaptor info

### ISSUE THREAD

**Issue by josephjclark** [OpenFn/kit#215]
When you call `openfn metata -a common -S { config }`, the metadata command will first check that it has a cached metadata result saved to its repo.

We have a little problem that the cache key is made only of the credential data. So if you take a credential and run it with adaptor `a`, it'll go ahead and run the function, cache the result, and return.

If you then run the same credential but ask to do it with adaptor `b`, or even `a@2.0.0`, the CLI will return you the previous cached result, and not in fact what the adaptor you asked for would have given you.

So basically the cache key needs to be a hash of `${adaptor-name}-${adaptor-version}-${credential'}`

TODO: we should also check that state.data doesn't affect the cache id.
